Py3 Compat

lmignon opened the pull request:
This PR makes the module compatible with Python 3 while remaining compatible with Python 2.7.

The first step in the migration process was the implementation of a continuous integration process based on travis and the refactoring of the tests. (see #10 for the motivations)

In a second step, the code was modified syntactically to be both Python 2 and Python 3 compliant.

The httplib2 library was then replaced by python-request (http://docs.python-requests.org/en/master/). It natively supports Python 2 and Python 3, manages sessions better (perf are improved), has no problem with large files and is very well maintained and documented. This library seems to be a de facto standard. 

Finally, the code is now flake8.

These changes constitute a major evolution of the library. Even if all the unit tests allow us to believe that no regression was introduced by this refactoring, these changes must be validated by an intensive use of the library.

lmignon committed:
Add travis for CI
THE CI use a docker image of Alfresco to be able to run the tests on a
clean install of Alfresco

lmignon committed:
Fix getContentStream service. Close #CMIS-1049

lmignon committed:
apply coverage on cmislib only

lmignon committed:
Cleanup tests

lmignon committed:
[FIX] BrowserBinding: Properly initialize the BrowserDocument returned by the setContentStream method

lmignon committed:
[FIX] DocumentTest: Fix testRelationship

lmignon committed:
[IMP] Move tests to py.test
Py.test allow us to run the same tests for a list of configurations. In one run we are therefore able to run the same tests for the browser binding and the atompub binding on alfresco. At this stage, the configurations is only defined for Alfresco, but it's easy to add configurations for others CMIS server.

lmignon committed:
travis: travis_wait

lmignon committed:
py3compat: replace 'has_key' with 'in' for dict

lmignon committed:
py3compat: 'print xyx' -> 'print(xyx)'

lmignon committed:
py3compat: minidom node attributes has_key compatibility

lmignon committed:
py3compat: import from right place if py2 or py3

lmignon committed:
py3compat: dict itervalues iteritems

lmignon committed:
py3compat: replace httplib2 by requests and fix StringIO->BytesIO

lmignon committed:
py3compat: more code shared between Atom and Browser

lmignon committed:
py3compat: flake8

x2764tech commented:
What would it take to get this into an official release?

lmignon commented:
@x2764tech We have to ask on the chemistry mailing list. dev@chemistry.apache.org
Do you have some feedback to provide on this new version? I use it in many projects without any problems but only the BrowserBinding part. I would be interested to know if other people have had the opportunity to test this new version.

x2764tech commented:
I'm planning on using the AtomPub part, to integrate with JupyterHub, so I'll feed back any issues I find there. 
Using `Session` for the requests makes a lot of things a lot simpler too.

lmignon committed:
BrowserBinding: quote literal '+' into url path
by default, python requests doesn't quote the sting literal '+'. Nevertheless this literal must be quoted when used in the getObjectByPath request

lmignon committed:
travis: use virtual machine in place of container

lmignon committed:
Fix new flake8 warning

lmignon committed:
adapt code for pytest>=4.0

lmignon committed:
skip test with '+' into folder name for alfresco < 5.2.0

benwillig committed:
[CHG] updated version number to 0.7.1
